test/cli/gateway_test.go

```diff
@@ -166,10 +166,10 @@ func TestGateway(t *testing.T) {
 		t.Parallel()
 		node.IPFS("name", "publish", "--allow-offline", cid)
 
-		t.Run("GET invalid IPNS root returns 400 (Bad Request)", func(t *testing.T) {
+		t.Run("GET invalid IPNS root returns 500 (Internal Server Error)", func(t *testing.T) {
```

Comment:
This one sgtm, consistent with https://github.com/ipfs/kubo/pull/9589 (`QmInvalid` is not a CID, but technically _could_ be a DNS name, (e.g. something from https://handshake.org/), so we can't say its Bad Request, could be DNS resolution error.)

Reply:
I'm slightly confused. This test, with the updates (https://github.com/ipfs/kubo/pull/9660#discussion_r1115664357) fails locally but passes on CI. I wonder if there's some weird distinction between the CI env (Linux) and my laptop (macOS) that makes the resolution fail at different points, raising different errors.

Reply:
Let's continue in https://github.com/ipfs/go-libipfs/issues/185